modules/S116_qemu_version_detection.sh

```diff
@@ -80,11 +81,20 @@ version_detection_thread() {
     #└─$ grep -a -o -H -E "Version: 1.8" /home/m1k3/firmware/emba_logs_manual/test_dir300/s115_usermode_emulator/qemu_tmp_radvd.txt                                                    130 ⨯
     # /home/m1k3/firmware/emba_logs_manual/test_dir300/s115_usermode_emulator/qemu_tmp_radvd.txt:Version: 1.8
     # /home/m1k3/firmware/emba_logs_manual/test_dir300/s115_usermode_emulator/qemu_tmp_radvd.txt:Version: 1.8
-    mapfile -t BINARY_PATHS < <(strip_color_codes "$(grep -a "Emulating binary:" "$LOG_PATH_" 2>/dev/null | cut -d: -f2 | sed -e 's/^\ //' | sort -u 2>/dev/null)")
+    for VERSION_DETECTED in "${VERSIONS_DETECTED[@]}"; do
+      mapfile -t LOG_PATHS < <(strip_color_codes "$(echo "$VERSION_DETECTED" | cut -d: -f1 | sort -u)")
+      for LOG_PATH_ in "${LOG_PATHS[@]}"; do
+        mapfile -t BINARY_PATHS_ < <(strip_color_codes "$(grep -a "Emulating binary:" "$LOG_PATH_" 2>/dev/null | cut -d: -f2 | sed -e 's/^\ //' | sort -u 2>/dev/null)")
+        for BINARY_PATH_ in "${BINARY_PATHS_[@]}"; do
+          # BINARY_PATH is the final array which we are using further
+          print_output "binary path: $BINARY_PATH_"
```

Comment:
This line seems unnecessary.
![image](https://user-images.githubusercontent.com/17053360/147575280-232c5fe8-a9a3-4aa0-b72e-ae8f8e120d04.png)